### PR TITLE
Pareto-smoothing updates

### PR DESCRIPTION
**Pull request by n-kall**
#### Summary

- Add functionality to `pareto_smooth` and related functions to operate on log weights (#313)
- Add functionality to `weight_draws` to Pareto-smooth weights (#313)
- Improve documentation for Pareto functions (#312)
- fix NA output (#315)
- change `r_eff` default to 1, rather than `NULL`

#### Discussion points

What should we do with khat when using `weight_draws`? Should we add a message or warning if khat > 0.7 (or khat_threshold)?

#### TODO
- add tests

#### Copyright and Licensing

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: R/weight_draws.R [hunk 1]
```diff
@@ -50,9 +52,12 @@ weight_draws <- function(x, weights, ...) {
 
 #' @rdname weight_draws
 #' @export
-weight_draws.draws_matrix <- function(x, weights, log = FALSE, ...) {
+weight_draws.draws_matrix <- function(x, weights, log = FALSE, pareto_smooth = FALSE, ...) {
   log <- as_one_logical(log)
   log_weights <- validate_weights(weights, x, log = log)
+  if (pareto_smooth) {
```
Comment (paul-buerkner) [inline R/weight_draws.R:58]: can you make sure that all new input arguments are checked for validity?

**Review by paul-buerkner — CHANGES_REQUESTED** [review]
PR looks good to me. I just have one request for changes.

File: R/weight_draws.R (hunk 1 above)
Comment (n-kall) [reply R/weight_draws.R:64]: I added `checkmate::expect_flag(pareto_smooth)` in each of the methods. Would this be ok?

**Comment by paul-buerkner** [comment]
I would prefer as_one_logical as I also use for the log argument.

github-actions[bot] ***@***.***> schrieb am Mi., 15. Nov.
2023, 17:50:

File: R/weight_draws.R (hunk 1 above)
Comment (paul-buerkner) [reply R/weight_draws.R:64]: Looks good!

**Comment by paul-buerkner** [comment]
Ready to merge from my side. Anything you want to change before I merge?

**Comment by n-kall** [comment]
I've now added tests and a message if pareto-k is high. I think it's ready from my side

**Pull request merged**